coil-base/src/main/java/coil/decode/BitmapFactoryDecoder.kt

```diff
@@ -162,7 +162,12 @@ internal class BitmapFactoryDecoder(private val context: Context) : Decoder {
         var outBitmap: Bitmap? = null
         try {
             outBitmap = safeBufferedSource.use {
-                BitmapFactory.decodeStream(it.inputStream(), null, this)
+                if (shouldDecodeFromByteArray(outMimeType)) {
```

Comment:
Can you inline this method here + add an `SDK_INT < 19` check? Thanks!

Reply:
I'm not entirely sure this issue never happens on any API 19 device, but everytime I saw it the MIME type was null so maybe as extra safety we could just fallback to byte array decoding every time we get a null MIME type no matter the API version or image format?

Reply:
Hmm I'd guess that it's probably constrained to < API 19 since there was a lot of work done to the graphics system in API 19. Also I'm not sure we should always fall back to byte array decoding if mime type is null since it has worse performance as it requires us to buffer the entire source into memory before decoding.

Reply:
Okay, I added the condition to only use byte array decoding on API < 19